File: backend/src/main/java/com/apihour/backend/service/VacationService.java
package com.apihour.backend.service;

import com.apihour.backend.model.Vacation;
import com.apihour.backend.repository.VacationRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;
import java.util.Optional;

@Service
public class VacationService {

  @Autowired
  private VacationRepository vacationRepository;

  public List<Vacation> getAllVacations() {
    return vacationRepository.findAll();
  }

  public List<Vacation> getVacationsByUserId(String userId) {
    return vacationRepository.findByUserIdOrderByStartDateDesc(userId);
  }

  public List<Vacation> getVacationsByUserIdAndStatus(String userId, Vacation.VacationStatus status) {
    return vacationRepository.findByUserIdAndStatus(userId, status);
  }

  public List<Vacation> getVacationsByUserIdAndDateRange(String userId, Date startDate, Date endDate) {
    return vacationRepository.findByUserIdAndStartDateBetween(userId, startDate, endDate);
  }

  public Optional<Vacation> getVacationById(String id) {
    return vacationRepository.findById(id);
  }

  public Vacation createVacation(Vacation vacation) {
    return vacationRepository.save(vacation);
  }

  public Vacation updateVacation(String id, Vacation vacationDetails) {
    Optional<Vacation> vacation = vacationRepository.findById(id);
    
    if (vacation.isPresent()) {
      Vacation existingVacation = vacation.get();
      existingVacation.setStartDate(vacationDetails.getStartDate());
      existingVacation.setEndDate(vacationDetails.getEndDate());
      existingVacation.setStatus(vacationDetails.getStatus());
      existingVacation.setReason(vacationDetails.getReason());
      
      return vacationRepository.save(existingVacation);
    }
    
    return null;
  }

  public boolean deleteVacation(String id) {
    Optional<Vacation> vacation = vacationRepository.findById(id);
    
    if (vacation.isPresent()) {
      vacationRepository.delete(vacation.get());
      return true;
    }
    
    return false;
  }
}

File: frontend/src/services/vacationService.js
import api from '../api/axios';

const vacationService = {
  getAllVacations: async () => {
    const response = await api.get('/vacations');
    return response.data;
  },

  getVacationsByUserId: async (userId) => {
    const response = await api.get(`/vacations/user/${userId}`);
    return response.data;
  },

  getVacationsByUserIdAndStatus: async (userId, status) => {
    const response = await api.get(`/vacations/user/${userId}/status/${status}`);
    return response.data;
  },

  getVacationById: async (id) => {
    const response = await api.get(`/vacations/${id}`);
    return response.data;
  },

  createVacation: async (vacationData) => {
    const response = await api.post('/vacations', vacationData);
    return response.data;
  },

  updateVacation: async (id, vacationData) => {
    const response = await api.put(`/vacations/${id}`, vacationData);
    return response.data;
  },

  deleteVacation: async (id) => {
    const response = await api.delete(`/vacations/${id}`);
    return response.data;
  }
};

export default vacationService;
